### PR TITLE
feat: add global yarn version, remove concept of last used version

### LINKED ISSUE #181: continually see yvm `No default yarn version set` even after install/use commands
**Describe the bug**

**To Reproduce**
First shell:
screen shot 2018-12-02 at 1 14 26 pm
New shell, shows it again (and env variable is set correctly):
screen shot 2018-12-02 at 1 16 12 pm

**Expected behavior**
Not to see `No default yarn version set` when default version is set.

**Screenshots**
(Added above)

**Desktop (please complete the following information):**
 - OS: Mac
 - YVM Release: https://github.com/tophat/yvm/releases/tag/2018-12-01T22-40-50.567Z

**Additional context**

### PR DESCRIPTION
**Pull request by jakebolam**
## Description
Motivation was originally to fix #181 + #180 . Found a better approach which works in a few more cases. 
Removes the need for #134 + #163 
Fixes #181 + #180 

This approach sets a global yvm yarn version. Then any command run will look for the yvm version to use as follows:
- Looks for version in command string
- Looks for rcfile
- Looks for global version

### DevQA Prep
- Install the latest version and .sh commands.

### DevQA Steps
- Open a new shell (Should see the please set default version)
- Follow the instructions and open a new terminal, should load and have a default version of yarn set
- If you have a .yarnrc in the folder your shell opened up in, will set to that yarn version

- All commands such as yvm use, yvm install etc will follow the approach of:
- Looks for version in command string
- Looks for rcfile
- Looks for global version

### Comments:
After this will create a ticket to update `yvm list` to show the default version

Also after this will create an issue for version validation.

### CODE REVIEW

**Comment by jakebolam** [comment]
@aeldamaty & @mcataford do you mind weighing in on this PR

File: src/yvm.js [hunk 1]
```diff
@@ -1,7 +1,12 @@
 const argParser = require('commander')
 
 const { ensureVersionInstalled } = require('./util/install')
-const { getSplitVersionAndArgs } = require('./util/version')
+const {
```
Comment (jakebolam) [inline src/yvm.js]: Reformatted this file with prettier. Let me know if you want this in a different PR

File: test/util/version.test.js [hunk 2]
```diff
@@ -0,0 +1,43 @@
...
+        expect(getDefaultVersion(mockYVMDir)).toEqual(mockVersion)
+    })
+
+    it('Returns no version if one is not set, when there is none', () => {
+        expect(getDefaultVersion(mockYVMDir)).toBeUndefined()
+    })
+})
+
+describe('yvm valid version', () => {
+    it('Valid versions', () => {
```
Comment (jakebolam) [inline test/util/version.test.js]: Super basic tests for version validation. There needs be a whole PR to fix the issues here

File: test/util/version.test.js [hunk 3]
```diff
@@ -0,0 +1,43 @@
...
+        expect(isValidVersionString('1.9.2')).toBe(true)
+    })
+
+    it('Invalid versions', () => {
+        expect(isValidVersionString('v1.9.2')).toBe(false)
```
Comment (jakebolam) [inline test/util/version.test.js]: Lots of these should ideally move to valid versions, once we have better interpretation of versions.
Will file an issue referencing this once this PR is merged

File: src/yvm.js (hunk 1 above)
Comment (mcataford) [reply src/yvm.js]: That's fine by me. 👍

File: src/util/version.js [hunk 4]
```diff
@@ -1,8 +1,43 @@
...
+}
+
+function getDefaultVersion(yvmPath = defaultYvmPath) {
+    const versionStoragePath = path.join(yvmPath, '.default-version')
+    try {
+        const versionJSONString = fs.readFileSync(versionStoragePath, 'utf8')
+        const versionJSON = JSON.parse(versionJSONString)
+        const version = versionJSON.defaultVersion
+        if (!version) {
+            throw new Error(`Version JSON exists but contains no key 'version'`)
```
Comment (aeldamaty) [inline src/util/version.js]: no key 'defaultVersion'

**Comment by aeldamaty** [comment]
One small comment, other than that CR and QA LGTM.

File: src/util/version.js (hunk 4 above)
Comment (jakebolam) [reply src/util/version.js]: @aeldamaty good catch, addressed

**Pull request merged**